src/main/java/de/hs_mannheim/informatik/ct/persistence/services/DynamicContentService.java

```diff
@@ -61,9 +64,24 @@
         return out.toByteArray();
     }
 
-    public void writeRoomsPrintOutDocx(List<Room> rooms, OutputStream outputStream, Function<Room, UriComponents> uriConverter) throws IOException, XmlException {
-        try(val document = getRoomsPrintOutDox(rooms, uriConverter)) {
-           document.write(outputStream);
+    public void writeRoomsPrintOutDocx(List<Room> rooms, ZipOutputStream outputStream, Function<Room, UriComponents> uriConverter) throws IOException, XmlException {
+        List<XWPFDocument> ListOfDocuments = getRoomsPrintOutDox(rooms, uriConverter, false);
+
+        int counter = 0;
+        outputStream.putNextEntry(new ZipEntry(rooms.get(0).getBuildingName() + "/"));
+        for (XWPFDocument document : ListOfDocuments) {
+
+            try {
+                ByteArrayOutputStream out = new ByteArrayOutputStream();
+                document.write(out);
+                out.close();
+                byte[] xwpfDocumentBytes = out.toByteArray();
+                outputStream.putNextEntry(new ZipEntry(rooms.get(0).getBuildingName() + "/" + rooms.get(counter).getName() + ".docx"));
```

Comment:
If you iterate over the rooms instead of the documents, you can just do `room.getBuildingName()` instead.